modules/phase_field/src/postprocessors/GrainTracker.C

```diff
@@ -1340,19 +1445,34 @@ GrainTracker::boundingRegionDistance(std::vector<MeshTools::BoundingBox> & bboxe
   return min_distance;
 }
 
+unsigned int
+GrainTracker::getNextUniqueID()
+{
+  /**
+   * Get the next unique grain ID but make sure to respect
+   * reserve ids.
+   */
+  while (++_max_curr_grain_id <= _reserve_grain_first_idx + _n_reserve_ops)
```

Comment:
I do agree, though, that the +1 should not be there (it is just not what your previous code did).